GRAPH.md

```diff
@@ -0,0 +1,631 @@
+  ModuleInstrumentation --> ModuleConstructor;
+  ModuleInstrumentation --> ModuleSourceBindingReflection;
+
+end
+
+```
+
+# Motivating Use-cases
+
+## Non-JavaScript not-host-defined modules
+
+Hosts may enable non-JavaScript modules to participate in the JavaScript module
+graph.
+Web hosts, for example, may enable JSON and WASM.
+If JavaScript were to enable user code to implement non-JavaScript module
+sources, these and other languages could participate on host implementations
+that did not anticipate these features, and allow user code to explore the
+range of useful module kinds ahead of host implementations, accelerating
+the evoluation of the JavaScript ecosystem.
+
+Of the current proposed language features, the language would need:
+***a minimal virtual module source protocol***.
+The minimal protocol would enable non-JavaScript languages that do not need
+JavaScript features like cyclic dependencies, lazy bindings, or early
+initialization of hoisted declarations.
+The ***maximal virtual module source protocol*** would be sufficient for
+emulating JavaScript in JavaScript and any other language that benefits from
+these behaviors.
+
+## Asset modules
+
+Asset modules are non-JavaScript not-host-defined modules as in the above
+motivating use case, but merit specific consideration since module source
+virtualization is sufficient to unlock their use cases, although other
+mechanisms are possible.
+
+Consider some possible non-JavaScript asset modules:
+
+- Binary modules could export a default array buffer.
+
+- Text modules could export decoded text as a default export.
+
+- An image module could export a default representation of an image, a blob
+  reference to an image, a URL, or a file system path to the image.
+  Incorporating images and similar resources in the JavaScript module graph
+  ensures that a program is portable between development and production
+  environments, allowing a JavaScript bundler to observe the dependency on an
+  image asset.
+
+- JSON modules test the boundary between a language and an asset or resource.
+  They do not have dependencies or side-effects.
+  Their behavior is much more like decoding than evaluation.
+
+- One valid interpretation of a CSS module would be to incorporate a
+  style sheet in the surrounding web page as a side effect of evaluation.
+  Such a module would export nothing but allow a bundler to observe the
+  dependency and ensure a topological order to the included sheets.
+  The module might also export a representation of the generated sheet as a
+  mutable resource.
+
+  Style sheets have their own notion of dependencies, and a valid
+  interpretation of a CSS module might subvert CSS imports and allow
+  CSS modules to benefit from JavaScript module resolution,
+  such that a bundler can relocate resources between development and
+  production.
+
+## Web Assembly not-host-defined modules
+
+Not all hosts necessarily implement the web's specifications for allowing
```

Comment:
```suggestion
Not all hosts necessarily implement the web's specifications for allowing
WebAssembly to participate in the JavaScript module graph.
As above, ***a minimal virtual module source protocol*** would allow
user code to emulate the standard behavior for WebAssembly.
```